### PR TITLE
Feature: ledger precision is configurable and enforced on transfer amounts

### PR DESCRIPTION
**Pull request by lumberj**
Scale and precision configurable
`LEDGER_AMOUNT_SCALE`
`LEDGER_AMOUNT_PRECISION`

Throw exception if amount precision > 15 when using oracledb

Exposed via metadata endpoint

### CODE REVIEW

**Comment by lumberj** [comment]
Related integration test change: https://github.com/interledger/five-bells-integration-test/pull/14

**Comment by MatthewPhinney** [comment]
LGTM (once all tests pass).

**Comment by lumberj** [comment]
@MatthewPhinney there's a dependent integration test PR https://github.com/interledger/five-bells-integration-test/pull/14

File: src/lib/config.js [hunk 1]
```diff
@@ -10,33 +10,47 @@ function isRunningTests () {
...
+}
+
+function parseAdminConfig () {
   const admin_user = Config.getEnv(envPrefix, 'ADMIN_USER') || 'admin'
   const admin_pass = Config.getEnv(envPrefix, 'ADMIN_PASS')
   const admin_fingerprint = Config.getEnv(envPrefix, 'ADMIN_TLS_FINGERPRINT')
 
   if (admin_pass || admin_fingerprint) {
-    localConfig.default_admin = _.omit({
+    return _.omit({
```
Comment (clark800) [inline src/lib/config.js]: I think this is missing a return for the else case

Comment (lumberj) [inline src/lib/config.js]: The `else` is that admin config is left undefined, same as original behavior

File: src/lib/config.js [hunk 2]
```diff
@@ -44,8 +58,34 @@ function loadConfig () {
       }
     }
   }
+}
+
+function validateConfig () {
+  // Validate precision
+  const commonConfig = Config.loadConfig(envPrefix)
+  const isOracle = commonConfig.get('db.uri').match(/oracle/) !== null
```
Comment (clark800) [inline src/lib/config.js]: Might be safer to test `startsWith('oracle://')`, the server name could contain "oracle".

File: src/lib/config.js (hunk 1 above)
Comment (clark800) [inline src/lib/config.js]: It would be more clear to have an explicit `return undefined`; otherwise the reader is wondering if it is intentional or not.

**Pull request merged**